### PR TITLE
Fix footer formatting issue

### LINKED ISSUE #24: [Bug] - Footer overlapping page content
### Bug Description:

Footer seems to be fixed to bottom of viewport, overlapping content.

---

### Reproduction Steps:

1. Open the website

---

### Expected Behavior:

If page content is shorter than screen, footer should be at base of screen.
If content is longer than screen, footer should be after content.

---

### Screenshots (If applicable):

Screenshot 2023-10-21 at 7 42 23 AM

---

### Possible Solution (optional):

Only give `fixed` position if content is less than screen height.

### PR DESCRIPTION
**Pull request by Jared-Krajewski**
## This PR:

Resolves #24

---

## Screenshots (if applicable):

Screenshot 2023-10-21 at 2 17 01 PM
 here.

---

## Future Steps/PRs Needed to Finish This Work (optional):

center socialLinks, fix roseLogo position, and clean up code.

### CODE REVIEW

**Comment by andycwilliams** [comment]
A definite improvement. A minor thing is that it has a small unnecessary scroll bar.

Capture3

Edit: Actually am testing other branches and the same issue is present. So it doesn't seem to be caused by the footer. Will have to test more.

---

Also, it's wonky with certain specific dimensions, apparently mainly with those that tablets use. This shows jumbled icons and text.

Capture

Changing `md` to `sm` on line 187 fixed this:

```
position: { xs: 'static', md: 'absolute' },
```

...although it also caused the text to overlap with the social media links. So it'll take more than that to resolve.

---

Whereas here it has the intended layout, just cut off. Worth noting that it also throws off the volunteer section, so that'll need addressing as well.

Capture2

---

I haven't tested everything, so I'm not a huge help with solutions right now. For the moment I just wanted to point out what my browser is showing.

---

Edit: Not a bug, but I think it might look nice to have the logo and `CODE PDX` more centered, namely in the first screenshot, so it's more even with the social media links. Maybe larger font as well.

Plus, we may want to change either the blob or logo color. Right now it's hard to distinguish the logo.

**Comment by Jared-Krajewski** [comment]
@andycwilliams  this is up as a draft/WIP to link it to issue #24 Scott created. I believe all of the things you mentioned have been fixed. Just trying to figure out something pushing icons to the right in larger viewports then this is ready for review. I will change the status when its ready. I have a separate issue for color theming and still need to put final colors into Figma then they can be added into the page.

**Comment by andycwilliams** [comment]
Cool cool. Just thought you wanted to work even more on the footer.

> I believe all of the things you mentioned have been fixed. Just trying to figure out something pushing icons to the right in larger viewports then this is ready for review.

Do you mean they're fixed in a branch you're working on but haven't yet pushed?

Anyway, in terms of addressing issue 24 this is good.

**Comment by Jared-Krajewski** [comment]
> Cool cool. Just thought you wanted to work even more on the footer.
> 
> > I believe all of the things you mentioned have been fixed. Just trying to figure out something pushing icons to the right in larger viewports then this is ready for review.
> 
> Do you mean they're fixed in a branch you're working on but haven't yet pushed?
> 
> Anyway, in terms of addressing issue 24 this is good.

Yes they are in an un-published commit.

**Comment by Jared-Krajewski** [comment]
The text still goes a little crazy near mobile view and will need to be fixed in the future but not worried for now because its also unlikely that anyone will be at that VW. Other issues should be addressed besides styling that will be changed after figma edits.

Edit: Andy, I did borders around all of the footer elements and the socialLinks were going past the body, so not sure why its happening in others for you but there was certainly an overflow from the footer that would have caused it. 

https://github.com/codeforpdx/codepdx_website/assets/114430929/b055617c-ae77-4a9c-b18d-da5d7a525309

**Comment by ogorman89** [comment]
Much better. 

A couple of questions:

Do we want the footer to stick to the bottom of the screen? Right now, it floats up to the content. 

Screen Shot 2023-10-22 at 8 29 59 AM

Visually, should we hide the CODE PDX in the mobile view? We still have "2023 CODE PDX" at the very bottom. I think a centered logo might look more symmetrical. 

Screen Shot 2023-10-22 at 8 36 47 AM

Links in the tablet section do appear to get jumbled, as Andy noted. Should we switch to the compact mobile view for tablet viewing? Things get a bit crowded and the mobile stack looks so clean 😎 There is also the issue of the logo color being too similar to the background. 

Screen Shot 2023-10-22 at 8 35 42 AM

I'm in favor of merging as is since the major bug is fixed.

**Review by ogorman89 — APPROVED** [review]
Resolves #24: see comments on potential enhancements.

**Comment by Jared-Krajewski** [comment]
@ogorman89 

We probably do want it down.  I think with it floating up to the contents there should be no issue since the page will be long enough to keep it at the bottom of the view but I could be wrong and will see if I can figure out a fix.

Ill check with Hugh/Flo and a few others if they have a preference for removing the text next to the logo in mobile view. I was thinking about actually keeping the legal links to the right and the rose to the left but under the socialLinks if that makes sense. I think I might build out a few of the select components that drastically change in mobile view on Figma so we can have more of a plan on mobile view.  I will try to get that done sooner than later.

I think its mentioned earlier in this PR but the colors aren't final. I need to put the theme into Figma and then implement that to our theme.js after its reviewed. The links will scramble at certain resolutions when you do responsive but if you set it to any of the smaller devices specifically I think there are no issues, If its causing issues we can change the breakpoint to a larger view like you suggested.

**Review by andycwilliams — APPROVED** [review]
I'm not seeing the text overlap with the links anymore, so that looks all fixed.

The text does still get jumbled in certain dimensions. However, it only seems to happen within a pretty small window (i.e. 602 width to about 745 width). I honestly don't know how many devices actually fall within that range. It doesn't get jumbled in any of the predefined dimensions when using devtools, including the tablet ones.

In mobile views, `CODE PDX` may look nicer when placed under the logo than side-by-side as Ian suggested. Symmetrical tends to be more pleasing, although it doesn't look bad as-is so it's not a real issue for me.

I also would rather the footer stick to the bottom on all viewports. But I agree we can wait until there's more content on each page, then see.

That said, I'll approve this now. It resolves the issue and any changes can wait for a future PR.

**Comment by xscottxbrownx** [comment]
> @ogorman89
> 
> We probably do want it down. I think with it floating up to the contents there should be no issue since the page will be long enough to keep it at the bottom of the view but I could be wrong and will see if I can figure out a fix.
> 
> Ill check with Hugh/Flo and a few others if they have a preference for removing the text next to the logo in mobile view. I was thinking about actually keeping the legal links to the right and the rose to the left but under the socialLinks if that makes sense. I think I might build out a few of the select components that drastically change in mobile view on Figma so we can have more of a plan on mobile view. I will try to get that done sooner than later.
> 
> I think its mentioned earlier in this PR but the colors aren't final. I need to put the theme into Figma and then implement that to our theme.js after its reviewed. The links will scramble at certain resolutions when you do responsive but if you set it to any of the smaller devices specifically I think there are no issues, If its causing issues we can change the breakpoint to a larger view like you suggested.

I had to do this with another project of mine, where I dynamically added `position: fixed` or not on the footer. I can try to find it in the code if needed.    Because when content is too short for screen, we want it fixed to bottom.  But when content is longer than screen, we do not.    Don't really want a frame of navbar and footer, with scrollable content in-between.

**Comment by Jared-Krajewski** [comment]
@andycwilliams  
 > In mobile views, `CODE PDX` may look nicer when placed under the logo than side-by-side as Ian suggested. Symmetrical tends to be more pleasing, although it doesn't look bad as-is so it's not a real issue for me.

It was like that and there was a comment asking it to be side by side so it was changed. Mobile Figma board will be made and design input will be done in a group session or through Figma feedback. 

@xscottxbrownx I have it written up on local but if you find it please send it over so I can compare how I did it (not sure my solution is the best), its messing up a blob but otherwise works, when I get that fixed I will push it before I merge this.

File: src/components/global/Footer.jsx [hunk 1]
```diff
@@ -36,34 +38,38 @@ const legalLinks = [
...
     href={href}
+    aria-label={`Check us out on ${ariaLabel}`}
     target="_blank"
     rel="noopener"
-    color="#000"
     sx={{
+      color: '#000',
       display: 'flex',
-      alignItems: 'center',
       padding: '25px',
```
Comment (xscottxbrownx) [inline src/components/global/Footer.jsx]: make this `padding` in `renderSocialLinks` only a `py: '25xp'`, then see comment on line 189.

File: src/components/global/Footer.jsx [hunk 2]
```diff
@@ -130,73 +149,48 @@ const logoBlobStyle = {
...
-            position: 'static', // Remove absolute positioning
-            transform: 'none' // Reset transform
-          }
-        }}
-      >
-        <Box sx={socialBlobStyle}>{renderSocialLinks}</Box>
-      </Box>
-      {/* this stack contains the legal links and hook to add padding on viewport scale down */}
+      {/* this box contains social links*/}
+      <Box display={'flex'}>{renderSocialLinks}</Box>
```
Comment (xscottxbrownx) [inline src/components/global/Footer.jsx]: Make this instead a `<Stack direction="row" spacing={{xs: 2, sm: 3}}>{renderSocialLinks}</Stack>`.     You could play with the spacing on `sm` if you want it bigger, but I didn't test any bigger than this.   `Stack` is flex by default.

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]
There is an issue on skinnier mobile devices (like galaxy fold) where the spacing between the social media icons is too large and messes up everything.   I made two comments that fix this.

I don't see any other issues.  No jumbled text/link/alignment, no scrollbars, etc.

**Comment by xscottxbrownx** [comment]
And merge into Dev instead yeah?

**Comment by Jared-Krajewski** [comment]
> And merge into Dev instead yeah?

I had it set to main so when anyone creates a branch they don't have that footer bug and we had no other changes in Dev (currently still don't) that would require a second pull request with the exact same content into main if i do that. so this just skips a PR with no other changes, but generally yes Dev instead. I guess for the sake of consistency I can change it.

**Review by xscottxbrownx — APPROVED** [review]

**Pull request merged**